frontend/src/components/professor/GameManagement/AddChapterModal/AddOrEditChapterModal.js

```diff
@@ -115,21 +136,36 @@ export function AddChapterModal({ showModal, setShowModal, refetchChapterList, i
                     return errors
                   }}
                   onSubmit={(values, { setSubmitting }) => {
-                    ChapterService.sendNewChapterData({
+                    const sentValues = {
                       name: values.name,
                       sizeX: values.sizeX,
                       sizeY: values.sizeY,
                       imageId: values.imageId,
                       posX: values.posX,
                       posY: values.posY
-                    })
-                      .then(() => {
-                        setSubmitting(false)
-                        setShowModal(false)
-                        setIsSuccessModalOpen(true)
-                        setErrorMessage('')
-                        refetchChapterList()
-                      })
+                    }
+
+                    const afterSendActions = () => {
+                      setSubmitting(false)
+                      setShowModal(false)
+                      setIsSuccessModalOpen(true)
+                      setErrorMessage('')
+                      if (afterSuccessfulChange) {
+                        afterSuccessfulChange()
+                      }
+                    }
+
+                    if (!chapterDetails) {
+                      ChapterService.sendNewChapterData(sentValues)
+                        .then(() => afterSendActions())
+                        .catch((error) => {
+                          setSubmitting(false)
+                          setErrorMessage(error.response.data.message)
+                        })
+                    }
+
+                    ChapterService.sendEditChapterData({ chapterId: chapterDetails.id, editionForm: sentValues })
+                      .then(() => afterSendActions())
                       .catch((error) => {
                         setSubmitting(false)
                         setErrorMessage(error.response.data.message)
```

Comment:
Cały ten kod też wyrzuć poza html. Z tego co wiem to złą praktyką jest pisanie tak dużo js'a w htmlu. Pojedyncze rzeczy spoko, ale tu jest sporo rzeczy. A już wywoływanie endpointów z htmla to na pewno bardzo zła praktyka. Także wrzuć to wyżej i pewnie opakuj w `useCallback`